mmhuman3d/data/data_converters/gta_human.py

```diff
@@ -0,0 +1,293 @@
+from mmhuman3d.models.builder import build_body_model
+from .base_converter import BaseConverter
+from .builder import DATA_CONVERTERS
+
+# TODO:
+# 1. camera parameters
+# 2. root align using mid-point of hips
+
+
+def perspective_projection(points: torch.Tensor, focal_length: float,
+                           camera_center: Tuple):
+    """This function computes the perspective projection of a set of points.
+
+    Input:
+        points (bs, N, 3): 3D points
+        focal_length (bs,) or scalar: Focal length
+        camera_center (bs, 2): Camera center
+    """
+    batch_size = points.shape[0]
+    K = torch.zeros([batch_size, 3, 3], device=points.device)
+    K[:, 0, 0] = focal_length
+    K[:, 1, 1] = focal_length
+    K[:, 2, 2] = 1.
+
+    K[:, 0, -1] = camera_center[0]
+    K[:, 1, -1] = camera_center[1]
+
+    points = torch.transpose(points, 1, 2)  # (bs, 3, N)
+    projected_points = torch.matmul(K, points)  # (bs, 3, N)
+    projected_points = torch.transpose(projected_points, 1, 2)  # (bs, N, 3)
+    projected_points = projected_points[:, :, :
+                                        2] / projected_points[:, :,
+                                                              2].unsqueeze(
+                                                                  -1
+                                                              )  # (bs, N, 2)
+
+    return projected_points
+
+
+@DATA_CONVERTERS.register_module()
+class GTAHumanConverter(BaseConverter):
+    """GTA-Human dataset `Playing for 3D Human Recovery' arXiv`2021 More
+    details can be found in the `paper.
+
+    <https://arxiv.org/pdf/2110.07588.pdf>`__.
+    """
+
+    focal_length = 1158.0337
+    camera_center = (960, 540)
+
+    device = torch.device(
+        'cuda') if torch.cuda.is_available() else torch.device('cpu')
+
+    smpl = build_body_model(
+        dict(
+            type='SMPL',
+            keypoint_src='smpl_54',
+            keypoint_dst='smpl_49',
+            model_path='data/body_models/smpl',
+            extra_joints_regressor='data/body_models/J_regressor_extra.npy')
+    ).to(device)
+
+    def convert(self, dataset_path: str, out_path: str) -> dict:
+        """
+        Args:
+            dataset_path (str): Path to directory where raw images and
+            annotations are stored.
+            out_path (str): Path to directory to save preprocessed npz file
+
+        Returns:
+            dict:
+                A dict containing keys video_path, smplh, meta, frame_idx
+                stored in HumanData() format
+        """
+        # use HumanData to store all data
+        human_data = HumanData()
+
+        body_pose_list = []
+        global_orient_list = []
+        betas_list = []
+        transl_list = []
+
+        image_path_list = []
+        bbox_xywh_list = []
+
+        keypoints_2d_gta_list = []
+        keypoints_3d_gta_list = []
+        keypoints_2d_list = []
+        keypoints_3d_list = []
```

Comment:
we usually use this formatting- https://github.com/open-mmlab/mmhuman3d/blob/0cb4a56576b7c69ec72f756b931b3a67674a9ba6/mmhuman3d/data/data_converters/surreal.py#L129-L137

Reply:
Modified!